presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -293,6 +320,15 @@ private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
                         "-12345",
                         -1.5,
                         1));
+        List<Map> rowColumn = ImmutableList.of(
+                namedStruct("tinyint_to_smallint", (short) -1, "tinyint_to_int", 2, "tinyint_to_bigint", -3L, "smallint_to_int", 100, "smallint_to_bigint", -101L, "int_to_bigint", 2323L, "bigint_to_varchar", "12345", "float_to_double", 0.5),
+                namedStruct("tinyint_to_smallint", (short) 1, "tinyint_to_int", -2, "tinyint_to_bigint", null, "smallint_to_int", -100, "smallint_to_bigint", 101L, "int_to_bigint", -2323L, "bigint_to_varchar", "-12345", "float_to_double", -1.5));
+        assertEqualsIgnoreOrder(queryResult.column(9), rowColumn, "row_to_row field is not equal");
+        assertEqualsIgnoreOrder(
+                queryResult.column(10).stream().map(o -> Arrays.asList((Object[]) ((PrestoArray) o).getArray())).collect(Collectors.toList()),
+                rowColumn.stream().map(ImmutableList::of).collect(Collectors.toList()),
+                "list_to_list field is not equal");
+        assertEqualsIgnoreOrder(queryResult.column(11), rowColumn.stream().map(map -> ImmutableMap.of(map.get("tinyint_to_int"), map)).collect(Collectors.toList()), "map_to_map field is not equal");
```

Comment:
Remove this block of code. Instead, write them out directly in the lines above (without any logic).

Reply:
Realized `assertThat` don't support `JAVA_OBJECT` directly. Has to extract each nested field. 

```
java.lang.RuntimeException: Unsupported sql type JAVA_OBJECT
	at io.prestodb.tempto.internal.query.QueryResultValueComparator.compare(QueryResultValueComparator.java:108)
	at io.prestodb.tempto.assertions.QueryAssert.isAnyValueEqual(QueryAssert.java:344)
	at io.prestodb.tempto.assertions.QueryAssert.rowsEqual(QueryAssert.java:334)
	at io.prestodb.tempto.assertions.QueryAssert.containsRow(QueryAssert.java:316)
	at io.prestodb.tempto.assertions.QueryAssert.contains(QueryAssert.java:176)
	at io.prestodb.tempto.assertions.QueryAssert.containsOnly(QueryAssert.java:207)
	at io.prestodb.tempto.assertions.QueryAssert.containsOnly(QueryAssert.java:219)
	at com.facebook.presto.tests.hive.TestHiveCoercion.doTestHiveCoercion(TestHiveCoercion.java:285)
	at com.facebook.presto.tests.hive.TestHiveCoercion.testHiveCoercionOrc(TestHiveCoercion.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```